### PR TITLE
Fix typos, grammar etc in *.md files

### PR DESCRIPTION
**Pull request by ndokos**
Mostyly typos and grammar fixes.

Clean up some incorrect statements as well, mostly by deleting them.

### CODE REVIEW

File: doc/CONTRIBUTING.md [hunk 1]
```diff
@@ -82,7 +81,7 @@ the report as well, and they have been updated accordingly.
 1. Make sure to add proper details to the Issue raised
 1. Upload screenshot(if possible) in dashboard issues
 1. Apply proper labels to the Issue
-1. Make sure to add Milestone and Project to the issue to mention specific release.
+1. XXXX Make sure to add Milestone and Project to the issue to mention specific release where the issue was encountered.
```
Comment (portante) [inline doc/CONTRIBUTING.md:84]: Typically, the project's release engineer adds the `Milestone` and `Project`, so we should probably drop this step entirely.

**Review by portante — CHANGES_REQUESTED** [review]
> In `agent/tools-scripts/README.md`, lines 53-56:
> 
> I haven't checked the code but this is a bit counter-intuitive: how
> is the post-processed stuff collected at the controller?

If you see the commit message for 23cc097e2857e8069c3a4c6ab485a27ed6c19d73, when PR #1248 landed, the post-processing steps only execute locally.  So once all the collected tool data is sent, then it can be post-processed.  Does that make sense?

> In `doc/CONTRIBUTING.md`, line 85:
> 
> item marked with XXXX: are Project/Milestone to be filled in when the
> issue is opened? Are they supposed to be set to the release where the
> issue was encountered or is that supposed to go in the text?

Typically the project owner sets milestones and projects for issues and PRs.  We should remove this entirely.

File: doc/CONTRIBUTING.md [hunk 2]
```diff
@@ -67,12 +67,11 @@ the report as well, and they have been updated accordingly.
 1. Write appropriate Pull Request Title stating the fix
   1. Use present tense (ex. Fixes, Changes, Fixing, Changing..)
 1. Reference the issue that the PR is fixing with **“Fixes #issue_number”** in the description
-1. Provide a detailed description (at least 50 - 70 characters) of the changes; if UI related, add screenshots or gif
+1. Provide a detailed description of the changes; if UI related, add screenshots
 1. Make sure that the branches can be automatically merged (otherwise rebase the PR with `master`) and then click the drop down next to, `Create pull request`, and select `Create draft pull request`
 ![Image](https://lh5.googleusercontent.com/V14SjFhimKYF1fH6TXMfaoZtDCj2ZH0d9USqe8YHyn0xOOVYekiXtx2CwdOQbSvxWPB6JVEfi4jSM_mjkSMaaI7voQNYQ8gDWntMhzCMbj3wrK3H4eCSEsdVq_XP_aZMdb5h9xU4)
 
 1. Assign the PR to yourself and add appropriate labels
-1. Add "**DO NOT MERGE**" label if the work is not needed to be merged or there is no agreement on the work yet
```
Comment (portante) [inline doc/CONTRIBUTING.md:75]: This is actually useful.  I have that on an example PR for the indexing work.

**Review by portante — CHANGES_REQUESTED** [review]

File: doc/CONTRIBUTING.md (hunk 2 above)
Comment (ndokos) [reply doc/CONTRIBUTING.md:75]: OK

**Review by portante — APPROVED** [review]

**Review by dbutenhof — APPROVED** [review]

**Pull request merged**